x/ethbridge/types/msgs.go

```diff
@@ -376,3 +365,51 @@ func (msg MsgUpdateWhiteListValidator) GetSigners() []sdk.AccAddress {
 
 	return []sdk.AccAddress{cosmosSender}
 }
+
+// NewMsgSetNativeToken is a constructor function for MsgSetNativeToken
+func NewMsgSetNativeToken(cosmosSender sdk.AccAddress, networkDescriptor oracletypes.NetworkDescriptor, nativeToken string) MsgSetNativeToken {
+	return MsgSetNativeToken{
+		CosmosSender:      cosmosSender.String(),
+		NetworkDescriptor: networkDescriptor,
+		NativeToken:       nativeToken,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgSetNativeToken) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgSetNativeToken) Type() string { return "set_native_token" }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgSetNativeToken) ValidateBasic() error {
+	if msg.CosmosSender == "" {
```

Comment:
We should do this validation check: 
`
	if len(msg.Signer) == 0 {
		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, msg.Signer)
	}
`